feat(max): conversations retrieve and list actions

skoob13 opened the pull request:
## Problem

Customers want to see the message history, but we don't have endpoints.

## Changes

- Add list and retrieve actions to the conversations API.
- Retrieve the latest messages in the serializer.
- Add user-defined cursor rules because everything was included in git by default.

## Does this work well for both Cloud and self-hosted?

No

## How did you test this code?

Unit tests

skoob13 committed:
feat(conversation): retrieve/list

- Added `ConversationSerializer` to manage conversation data and messages.
- Updated `ConversationViewSet` to use the new serializer and handle message retrieval.
- Modified queryset filtering to allow message creation for any conversation while restricting retrieval to those with titles.
- Enhanced tests to cover new conversation creation scenarios and ensure proper filtering of assistant conversations.
- Updated `.gitignore` to ignore user-specific Cursor rules.

skoob13 committed:
fix: greptile

skoob13 committed:
fix: mypy

Twixes commented on ee/api/conversation.py:
```diff
@@ -35,15 +38,39 @@ def validate(self, data):
+class ConversationSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Conversation
+        fields = "__all__"
```
Let's be explicit with the fields, because that way we can improve in two ways:
1. We can omit `team_id`, which would just be duplicated from the endpoint path
2. We can explicitly make all the fields read-only (I know the endpoint doesn't have PATCH, but it will be much harder to accidentally allow something wrong in the future, for free)
```suggestion
        fields = ["user_id", "status", "type", "messages"]
        read_only_fields = fields
```
Also this exposes that we're including `user_id`, which in itself _probably_ won't be useful on the frontend, in that `UserBasicSerializer` will likely be way easier to consume in practice.

skoob13 replied:
Reasonable! I've updated the fields.

skoob13 committed:
fix(conversation): update ConversationSerializer fields

- Changed `fields` in `ConversationSerializer` from `"__all__"` to a specific list: `["id", "status", "type", "messages"]`.
- Set `read_only_fields` to the same list for clarity.

skoob13 committed:
fix: type -> title
